varinfo/cf_config.py

```diff
@@ -151,14 +147,7 @@ def _process_cf_item(
             # to all variables (see ICESat2 dimensions override, SPL4.* and
             # SPL3FTA grid_mapping overrides)
             pattern = cf_item['Applicability'].get('Variable_Pattern', '.*')
-
-            if 'Attributes' in cf_item:
-                results[pattern] = self._create_attributes_object(cf_item)
-
-            cf_references = cf_item.get('Applicability_Group', [])
-
-            for cf_reference in cf_references:
-                self._process_cf_item(cf_reference, results, mission, short_name)
+            results[pattern] = self._create_attributes_object(cf_item)
```

Comment:
I think this is the single change that I want to call attention to. The rest is mainly flattening the contents of the configuration file.

Now that the configuration file has changed, the [required attributes in an override or a supplement](https://github.com/nasa/earthdata-varinfo/pull/33/files#diff-cb2e38ad70c1243f56366dec874a890b06e7be15cf31b54e8ad770ba93a8cc98R323) are both the `Applicability` and the `Attributes` (where as before it was the `Applicability` and one of either the `Attributes` or `Applicability_Group`).

Ignoring all the stuff about grabbing and iterating through the `Applicability_Group`, the other change is that I've cut out the condition `if 'Attributes' in cf_item`, where it's now a required property. I mainly want someone to sanity check that makes sense (and that I've done it properly).

Reply:
This tracks to me.